### PR TITLE
Add react-datepicker definitions

### PR DESCRIPTION
**Pull request by kangax**
Fixes https://github.com/Hacker0x01/react-datepicker/issues/1306

### CODE REVIEW

File: definitions/npm/react-datepicker_v1.4.x/flow_v0.84.x-/react-datepicker_v1.4.x.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// Based on https://github.com/Hacker0x01/react-datepicker/blob/master/docs/datepicker.md
+
+declare module 'react-datepicker' {
+  declare type moment$Moment = Object;
```
Comment (kangax) [inline definitions/npm/react-datepicker_v1.4.x/flow_v0.84.x-/react-datepicker_v1.4.x.js]: Is there a way to reference the actual `moment$Moment` definition from the repo? I've done what rsuite_v3... does but aliasing `Moment` to `Object` is obviously pretty bad.

Comment (kangax) [reply definitions/npm/react-datepicker_v1.4.x/flow_v0.84.x-/react-datepicker_v1.4.x.js]: /cc @villesau

Comment (kangax) [reply definitions/npm/react-datepicker_v1.4.x/flow_v0.84.x-/react-datepicker_v1.4.x.js]: Ah... thanks @villesau, I'll read through those and subscribe

**Comment by kangax** [comment]
@villesau been meaning to add a comment next to `Object` with the link to those PR's but no big deal. I'm hesitant to duplicate the entire moment API here.

File: definitions/npm/react-datepicker_v1.4.x/flow_v0.84.x-/react-datepicker_v1.4.x.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    scrollableYearDropdown?: boolean,
+    selected?: Date,
+    selectsEnd?: boolean,
+    selectsStart?: boolean,
+    shouldCloseOnSelect?: boolean,
+    showMonthDropdown?: boolean,
+    showTimeSelect?: boolean,
+    showWeekNumbers?: boolean,
+    showYearDropdown?: boolean,
+    startDate?: Date,
```
Comment (mull) [inline definitions/npm/react-datepicker_v1.4.x/flow_v0.84.x-/react-datepicker_v1.4.x.js]: This seems incorrect? Providing a `Date` type for me yields `TypeError: date.clone is not a function`

**Pull request merged**